A78-grpc-metrics-wrr-pf-xds.md

```diff
@@ -0,0 +1,154 @@
+A78: gRPC OTel Metrics for WRR, Pick First, and XdsClient
+----
+* Author(s): @markdroth
+* Approver: @ejona86, @dfawley
+* Status: {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2024-03-22
+* Discussion at: https://groups.google.com/g/grpc-io/c/A2Mqz8OMDys
+
+## Abstract
+
+This document proposes some new metrics that will be added in gRPC for the
+Weighted Round Robin (WRR) and Pick First LB policies and for the XdsClient.
+It also adds a new optional label for the existing per-call metrics.
+
+## Background
+
+gRPC recently added a set of basic per-call metrics, defined in [A66].
+[A79] is building upon that by providing a framework for non-per-call
+metrics.  The metrics described in this document will be the first
+metrics added using that new non-per-call metric framework.
+
+### Related Proposals: 
+* [A66]: OpenTelemetry Metrics
+* [A79]: gRPC Non-Per-Call Metrics Framework (pending)
+* [A58]: Weighted Round Robin LB Policy
+* [A62]: Pick First: Sticky TRANSIENT_FAILURE and Address Order Randomization
+* [A27]: xDS-Based Global Load Balancing
+* [A28]: xDS Traffic Splitting and Routing
+* [A71]: xDS Fallback
+* [A57]: XdsClient Failure Mode Behavior
+
+[A66]: A66-otel-stats.md
+[A79]: https://github.com/grpc/proposal/pull/421
+[A58]: A58-client-side-weighted-round-robin-lb-policy.md
+[A62]: A62-pick-first.md
+[A27]: A27-xds-global-load-balancing.md
+[A28]: A28-xds-traffic-splitting-and-routing.md
+[A71]: A71-xds-fallback.md
+[A57]: A57-xds-client-failure-mode-behavior.md
+
+## Proposal
+
+This document proposes changes to the following gRPC components.
+
+### Optional xDS Locality Label
+
+When xDS is used, it is desirable for some metrics to include an optional
+label indicating which xDS locality the metrics are associated with.
+We want to provide this optional label for the metrics in both the
+existing per-call metrics defined in [A66] and in the new metrics for
+the WRR LB policy, described below.
+
+#### Per-Call Metrics
+
+To support the locality label in the per-call metrics, we will provide
```

Comment:
I think if the label is enabled, we should always add it.  I'd fine setting it to the empty string.

Reply:
I think either "" or "unknown" is fine. Let's add that language?

Reply:
Also, I think we need to specify which metrics this will show up on - 
* grpc.client.attempt.duration
* grpc.client.attempt.sent_total_compressed_message_size
* grpc.client.attempt.rcvd_total_compressed_message_size

Reply:
Added those details, as well as addressing your suggestion from #36049.